app/src/main/java/nu/yona/app/ui/LaunchActivity.java

```diff
@@ -118,10 +122,71 @@ else if (!getSharedUserPreferences().getBoolean(PreferenceConstant.STEP_PASSCODE
 		}
 		else if (!TextUtils.isEmpty(getSharedUserPreferences().getString(PreferenceConstant.YONA_PASSCODE, "")))
 		{
-			startNewActivity(bundle, YonaActivity.class);
+			checkForUserEntityVersion();// validate any updates to user entity before moving to next screen.
+		}
+	}
+
+	private void checkForUserEntityVersion()
+	{
+		try
+		{
+			JSONObject userJsonObj = APIManager.getInstance().getAuthenticateManager().getStoredUserObjectFromDB();
+			if (userJsonObj.isNull("version") || userJsonObj.getInt("version") != AppConstant.USER_ENTITY_VERSION)
+			{
+				if (!AppUtils.isNetworkAvailable(YonaApplication.getAppContext()))
```

Comment:
Extract this to a method like this:
```
private void reloadUserFromServer(String url)
{
	if (!AppUtils.isNetworkAvailable(YonaApplication.getAppContext()))
	{
		showMandatoryUserFetchError();
		return;
	}

	getUserFromServer(url);
}
```

Which you then can call as follows:
```
reloadUserFromServer(user.getLinks().getSelf().getHref());
```

Reply:
Done.